### ISSUE TITLE
OPL3 MIDI panning reversed (reverse stereo)

### ISSUE THREAD

**Issue by sneakernets** [chocolate-doom/chocolate-doom#562]
In D_EVIL for Doom 2, the gunshot pans are reversed (right-to-left instead of left-to-right).

Be careful, as Prince threw in some right-to-left "marching" to mix it up, but the beginning should all be left-to-right.

Also, when it comes to percussion, DMX doesn't change panning on percussion instruments until the next note. One gunshot note shouldn't be able to pan L-C-R at all.

Edit: Actually, the stereo seems to be reversed! D_ULTIMA is a good example. also has issues of panned notes switching sides in OPL3 mode!

**Comment by sneakernets** [comment]
@khokh2001 can you check on this one?

**Comment by sneakernets** [comment]
I fixed this locally and will submit a pull request if desired.
here is the code I modified in opl_sdl.c:

```c
static void FillBuffer(int16_t *buffer, unsigned int nsamples)
{
    unsigned int i;

    // This seems like a reasonable assumption.  mix_buffer is
    // 1 second long, which should always be much longer than the
    // SDL mix buffer.

    assert(nsamples < mixing_freq);

    if (opl_opl3mode)
    {
        Chip__GenerateBlock3(&opl_chip, nsamples, mix_buffer);

        // Mix into the destination buffer, doubling up into stereo.

        for (i=0; i<nsamples; ++i)
        {
            buffer[i * 2 + 1] = (int16_t) mix_buffer[i * 2];
            buffer[i * 2] = (int16_t) mix_buffer[i * 2 + 1];
        }
    }
    else
    {
        Chip__GenerateBlock2(&opl_chip, nsamples, mix_buffer);

        // Mix into the destination buffer, doubling up into stereo.

        for (i=0; i<nsamples; ++i)
        {
            buffer[i * 2] = (int16_t) mix_buffer[i];
            buffer[i * 2 + 1] = (int16_t) mix_buffer[i];
        }
    }
}
```

This is the equivalent of reversing the audio cables to a stereo.

**Comment by fragglet** [comment]
Fixed in 41eebca5614912abe656e443668b11214c6d51c1 (see context in #586).
